### PR TITLE
Make repo a single package with cluster support

### LINKED ISSUE #39: Make repo a single package with cluster support
This repo is currently setup as a monorepo using lerna and consists of two packages (srcmap-reverse-server and srcmap-reverser). The point of having it this way was to offload the process of sourcemapping individual lines in a way that would scale. However, since we want to bundle this as a single process that will run as a daemon it doesn't make sense to publish two different bundles that must be run in the background. Instead, this repo should be a single entity (without lerna). A node cluster will be used to scale the processing of individual lines.

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #39.

This repo is currently setup as a monorepo using lerna and consists of
two packages (srcmap-reverse-server and srcmap-reverser). The point of
having it this way was to offload the process of sourcemapping
individual lines in a way that would scale. However, since we want to
bundle this as a single process that will run as a daemon it doesn't
make sense to publish two different bundles that must be run in the
background. Instead, this repo should be a single entity (without
lerna). A node cluster will be used to scale the processing of
individual lines.

- Integration test passing
- Integrated cluster to scale srcmap reverse process of individual lines
- Have a few more tests that I need to get passing but I wanted to get
  what I have committed.

Signed-off-by: Will Johnson <william.c.johnson@intel.com>

### CODE REVIEW

File: source/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const server = http.createServer(
+    (request: http.IncomingMessage, response: http.ServerResponse) => {
+      const r: HighlandStreamT<Buffer> = highland(request);
+
+      r
+        .map(x => x.toString('utf-8'))
+        .collect()
+        .map((x: string[]) => x.join(''))
+        .flatMap((x: string) => {
+          const { trace }: { trace: string } = JSON.parse(x);
```
Comment (jgrund) [inline source/index.js]: If you moved the trace up into it's own map, this could all be points-free.

File: source/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  process.send({ setup: true });
+} else {
+  process.on('message', msg => {
+    highland([msg])
+      .otherwise(highland(['']))
+      .through(reverser)
+      .collect()
+      .errors((e: Error, push: Function) => {
+        process.send({ error: e });
+        push(e);
```
Comment (jgrund) [inline source/index.js]: Why both send and push?

File: package.json [hunk 3]
```diff
@@ -3,21 +3,21 @@
   "version": "0.0.0",
   "description": "Reverses a minified stack trace using a source map.",
   "private": true,
+  "main": "dist/srcmap-reverse.js",
   "scripts": {
-    "pretest": "lerna run clean; lerna bootstrap; lerna run prepublish",
-    "test": "NODE_ENV=test; jest --silent --projects packages/srcmap-reverse packages/srcmap-reverser",
+    "pretest": "yarn rollup",
+    "test": "NODE_ENV=test; jest",
```
Comment (johnsonw) [inline package.json]: I'm fine either way. It's kind of annoying if your debugging tests though. If you want I can take it out.

File: source/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        })
+        .map(xs => JSON.stringify(xs))
+        .pipe(response);
+    }
+  );
+
+  const port: number = +process.env.npm_package_config_port;
+  server.listen(port);
+
+  process.send({ setup: true });
```
Comment (johnsonw) [inline source/index.js]: Once the server is setup we send a notification to any process that's running it (the integration test is mainly what uses this). If we spin up srcmap-reverse and immediately send data to it before the http-server is setup we will get a socket error.

File: source/index.js (hunk 2 above)
Comment (johnsonw) [inline source/index.js]: I may have had this in while I was debugging. We only need to push so i'll remove the send.

File: package.json [hunk 5]
```diff
@@ -3,18 +3,20 @@
...
     "eslint": "eslint ./",
-    "bootstrap": "lerna bootstrap",
-    "clean": "lerna clean; rm -rf packages/srcmap-reverse/dist; rm -rf packages/srcmap-reverser/dist",
-    "rollup": "rollup -c rollup-config.js"
+    "rollup": "rollup -c rollup-config.js -i source/index.js -o $npm_package_main"
   },
   "pre-commit": ["flow", "eslint", "test"],
   "author": "IML Team",
+  "config": {
+    "port": 8082
```
Comment (jgrund) [inline package.json]: Is this needed?

**Pull request merged**